### PR TITLE
Implement setup=true behaviour for Agents page

### PR DESCRIPTION
**Pull request by ticky**
Spot the difference;

screen shot 2016-12-05 at 4 15 21 pm
screen shot 2016-12-05 at 4 15 32 pm

👻

### CODE REVIEW

**Comment by ticky** [comment]
This doesn’t have the `AgentConnectedModal` in it yet!

**Comment by keithpitt** [comment]
@ticky oh right! We have some `Dialog` technology you can use!!

**Comment by ticky** [comment]
## Now with dialogs:

### Old
old and busted

### New
new hotness

**Comment by keithpitt** [comment]
Oh nice! Does this mean it's ready for a code review party?

**Comment by ticky** [comment]
Yep! Go for it 👍

File: app/components/agent/Index/installation.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    }).isRequired,
+    relay: React.PropTypes.object.isRequired
+  };
+
+  state = {
+    isDialogOpen: false
+  };
+
+  componentDidMount() {
+    this._agentRefreshInterval = setInterval(this.fetchUpdatedData, 5::seconds);
```
Comment (keithpitt) [inline app/components/agent/Index/installation.js]: Maybe we should use `setTimeout` and then schedule another timeout once that's finished? Something like:

```
function refresh() {
  setTimeout(function() {
    this.props.relay.forceFetch(true, function() {
      this.refresh();
    })
  }, 5::seconds)
}
```

If requests are taking a long time (for some reason) they'll keep piling up, where as with this approach, it'll only schedule the next one after the first one finishes, if that makes sense?

File: app/components/shared/Dialog.js [hunk 2]
```diff
@@ -43,60 +110,71 @@ class Dialog extends React.Component {
...
-        <span className="block fixed flex items-center justify-center" style={{ top: 0, left: 0, bottom: 0, right: 0, zIndex: 1000 }}>
-          <ReactCSSTransitionGroup transitionName="transition-slide-up" transitionEnterTimeout={150} transitionLeaveTimeout={300}>
-            {this.renderDialog()}
-          </ReactCSSTransitionGroup>
-          {this.renderBackdrop()}
-        </span>
-      );
-    } else {
-      return null;
+  maybeClose = (event) => {
```
Comment (ticky) [inline app/components/shared/Dialog.js]: `closeMeMaybe`

Comment (keithpitt) [inline app/components/shared/Dialog.js]: Probably the best method name ever?

**Comment by ticky** [comment]
We haven’t used `setTimeout` in some other places which poll, so I think that’s a thing we should probably update all at once if we wanna be changing it!

**Comment by keithpitt** [comment]
Meh, it just something I noticed, but in reality I don't think it'd be an issue!

File: app/components/shared/Dialog.js (hunk 2 above)
Comment (ticky) [inline app/components/shared/Dialog.js]:
```javascript
/* heyNowIJustMetYouAndThisIsCrazyButHeresMyCallbackSo*/closeMeMaybe
```

**Comment by ticky** [comment]
It’s definitely a valid criticism, though! I’ll open an issue for it :)

**Pull request merged**